File: Sorting/Ruby/SelectionSort.rb
def swap(arr,i,j)
	temp = arr[i]
	arr[i] = arr[j]
	arr[j] = temp
	return arr
end # end func

def findSmallest(arr,start_pt)
    smallest = start_pt
    for i in (start_pt + 1)..arr.length-1 do
        if arr[i]< arr[smallest]
            then smallest = i
        end # end if
    end # end for
    return smallest
end # end func

def selectionSort(arr)
    for i in 0..arr.length-1 do
        smallest = findSmallest(arr, i)
        arr = swap(arr, smallest, i)
    end # end for
    return arr
end # end func

File: Sorting/Java/SelectionSort.java

public class SelectionSort implements SortingAlgorithm {
	protected void swap(int[] a, int i, int j){
		int temp = a[i];
		a[i] = a[j];
		a[j] = temp;
	}
	
	protected int smallest(int[] a, int start){
		int smallest = start;
		for (int i = start +1; i<a.length;i++){
			if (a[i]<a[smallest]){
				smallest = i;	
			}
		}
		return smallest;
	}
	
	public void sort(int [] a){
		for (int i=0; i<a.length;i++){
			int smallest = smallest(a,i);
			swap(a,smallest,i);
		}
	}
}
